apps/api/src/app/notification-template/usecases/delete-notification-template/delete-notification-template.usecase.ts

```diff
@@ -65,6 +68,18 @@ export class DeleteNotificationTemplate {
           changeId: parentChangeId,
         })
       );
+
+      for (const step of item.steps) {
+        await this.deleteMessageTemplate.execute(
+          DeleteMessageTemplateCommand.create({
+            organizationId: command.organizationId,
+            environmentId: command.environmentId,
+            userId: command.userId,
+            messageTemplateId: step._templateId,
+            parentChangeId: parentChangeId,
```

Comment:
Do we want to stop the execution of the notification template deletion if any of the message template deletions fail?